backend/de.metas.business/src/main/java/de/metas/invoice/invoiceProcessingServiceCompany/InvoiceProcessingServiceCompanyConfigMap.java

```diff
@@ -68,15 +74,40 @@ public InvoiceProcessingServiceCompanyConfigMap(@NonNull final List<InvoiceProce
 		}
 	}
 
+	/**
+	 * @return if there is an {@link InvoiceProcessingServiceCompanyConfig} with the given {@code customerId} (the one for which a service-company handles payments) at the given {@code validFrom}, then it is returned.<br>
+	 * If there is an older config for the given {@code customerId}, but a more recent config which does not have the given {@code customerId},
+	 * then {@link Optional#empty()} is returned.
+	 */
+	@NonNull
 	public Optional<InvoiceProcessingServiceCompanyConfig> getByCustomerIdAndDate(@NonNull final BPartnerId customerId, @NonNull final ZonedDateTime validFrom)
 	{
-		final ImmutableList<InvoiceProcessingServiceCompanyConfig> configs = bpartnersToConfigsSorted.get(customerId);
-		for (int i = configs.size() - 1; i >= 0; i--)
+		final ImmutableList<InvoiceProcessingServiceCompanyConfig> configsForCustomers = bpartnersToConfigsSorted.get(customerId);
+		for (int i = configsForCustomers.size() - 1; i >= 0; i--)
 		{
-			final InvoiceProcessingServiceCompanyConfig config = configs.get(i);
-			if (config.getValidFrom().isBefore(validFrom) || config.getValidFrom().isEqual(validFrom))
+			final InvoiceProcessingServiceCompanyConfig config = configsForCustomers.get(i);
+			final boolean configIsValid = config.getValidFrom().isBefore(validFrom) || config.getValidFrom().isEqual(validFrom);
+			if (configIsValid)
 			{
-				return Optional.of(config);
+				final ImmutableList<InvoiceProcessingServiceCompanyConfig> configsForCompanyBPartners = companyBPartnersToConfigsSorted.values().asList();
+				for (int j = configsForCompanyBPartners.size() - 1; j >= 0; j--)
+				{
+					final InvoiceProcessingServiceCompanyConfig companyConfig = configsForCompanyBPartners.get(j);
+					final boolean companyConfigIsValid = companyConfig.getValidFrom().isBefore(validFrom) || companyConfig.getValidFrom().isEqual(validFrom);
+					if (companyConfigIsValid)
+					{
+						if (config.equals(companyConfig))
+						{
+							return config.isBPartnerDetailsActive(customerId)
```

Comment:
question: since the config was fetched by "bpartnersToConfigsSorted.get(customerId)",
isn't config.isBPartnerDetailsActive(customerId) always true?